### PR TITLE
Usability features and UI polish

### PR DESCRIPTION
**Pull request by d11e9**
I propose some improvements I understand that some of their merits may be entirely subjective but here goes anyway.
- Cache successfully compiled contracts in localStorage (prevent loss of work)
- Show compile errors inline
- Allow resizing of the editor and output areas.
- General (highly subjective UI tidy / improvements )

feedback welcome, give it a test at: https://d11e9.github.io/browser-solidity/

### CODE REVIEW

**Comment by ghost** [comment]
:+1: for caching of solidity code (I assume you mean uncompiled contracts, @d11e9 ?)

**Comment by chriseth** [comment]
Wow, that is awesome! Thanks for contributing this!
I have some small comments (see below / inline) and one request:
As you may have noticed, this repository does not have a license file (and the files do not even have license headers). Are you fine with licensing your contribution with either MIT or GPL?

File: ballot.sol.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+var multi = function(func) { return func.toString().match(/[^]*\/\*([^]*)\*\/\}$/)[1]; }
```
Comment (chriseth) [inline ballot.sol.js:1]: This gives me small headaches. Is there no other way to do it? Is it portable?

File: index.html [hunk 2]
```diff
@@ -12,67 +12,206 @@
...
 }
-#output {
+
+#righthand-panel {
     position: absolute;
-    top: 120px;
-    left: 720px;
-    right: 10px;
+    top: 0;
+    width: 37em;
```
Comment (chriseth) [inline index.html:37]: Is that large enough? On my screen, I have about 3/4 editor and 1/4 righthand panel, and I would actually prefer the editor size to be at least 90 chars (about the position of the vertical line in the editor) and the righthand panel at least 1/3 of the screen or something like that.
But that is only a minor thing.

File: index.html [hunk 3]
```diff
@@ -80,366 +219,415 @@
...
+           var main = $('#righthand-panel');
+           var ghostbar = $('<div id="ghostbar">', {
+                css: {
+                    top: main.offset().top,
+                    left: main.offset().left
+                }
+            }).prependTo('body');
+           
+            $(document).mousemove(function(e){
+              ghostbar.css("left",e.pageX+2);
```
Comment (chriseth) [inline index.html:292]: Suggestion for another feature: Could we store that in localstorage?

File: solidity.svg [hunk 4]
```diff
@@ -1,64 +1,27 @@
...
-	<path fill="#010101" d="M2242.7,1613.8c2.6,0,3.9,1.3,3.9,3.9c0,0.2-0.7,2.2-2.2,6.1c-1.5,3.9-3.4,8.9-5.7,15.1
-		c-2.3,6.2-5,13.3-8,21.2c-3,7.9-6.2,16.1-9.6,24.5c-3.3,8.4-6.6,16.7-9.7,24.8c-3.1,8.1-6.1,15.6-8.7,22.3
-		c-2.7,6.7-4.9,12.5-6.8,17.2c-1.8,4.7-3,7.7-3.6,9c-0.9,1.7-2.1,2.5-3.6,2.5c-0.9,0-1.8-0.4-2.6-1.2c-0.8-0.8-1.2-1.7-1.2-2.6
-		c0-0.4,0.5-1.8,1.4-4.3c0.9-2.5,2-5.5,3.3-9c1.3-3.5,2.8-7.3,4.4-11.5c1.7-4.2,3.2-8.1,4.7-11.9c1.5-3.8,2.8-7.1,3.9-9.8
-		c1.1-2.8,1.8-4.6,2.2-5.5c-0.2-0.2-1.2-2.4-3-6.6c-1.8-4.2-4.1-9.6-6.8-15.9c-2.7-6.4-5.5-13.3-8.5-20.6c-3-7.4-5.7-14.3-8.3-20.6
-		c-2.6-6.4-4.8-11.7-6.5-16.1c-1.8-4.3-2.6-6.6-2.6-6.8c0-2.6,1.3-3.9,3.9-3.9c1.8,0,3,0.7,3.6,2.2l32.1,78.4l30.5-78.1
-		C2239.6,1614.6,2240.8,1613.8,2242.7,1613.8z"/>
-</g>
-</svg>
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (chriseth) [inline solidity.svg:1]: What did you change here?

Comment (d11e9) [inline solidity.svg:1]: the svg had a bunch of white space around it i cropped the image so that the svg size = icon size

File: index.html (hunk 3 above)
Comment (d11e9) [inline index.html:292]: Sure can, does this feature help with your feedback one previous, re default sizes?

File: index.html (hunk 2 above)
Comment (d11e9) [inline index.html:37]: See comment below about caching editor sizes.

File: ballot.sol.js (hunk 1 above)
Comment (d11e9) [inline ballot.sol.js:1]: yeah it certainly is a hack, but the only way i could remove it from index.html, without having to concat tonnes of strings together.

reference: http://stackoverflow.com/a/5571069/371040 (includes **claimed** browser support)

**Comment by d11e9** [comment]
rebased onto updated `gh-pages`

**Comment by d11e9** [comment]
Added caching editor resizing d74fb4fdd512c14eab3667e535ac75040d0428b2

File: index.html (hunk 3 above)
Comment (d11e9) [inline index.html:292]: feature added to PR.

**Comment by d11e9** [comment]
MIT Suites me. You need me to add the License file?

File: index.html (hunk 2 above)
Comment (d11e9) [inline index.html:37]: Also worth noting is that the 37em applies to the output panel on the right, and the editor would (in the default case take up the rest of the space)

File: ballot.sol.js (hunk 1 above)
Comment (chriseth) [inline ballot.sol.js:1]: Ok, so I think it is fine. If it fails, we do not have example code which is not too bad.

**Comment by chriseth** [comment]
I am not particularly keen on having the "fork me on github" banner, as it loads external content (everything else is local, I think) and it is quite a large distraction. I think it is nice to have on a landing or documentation page, but not as part of an application.

Please add the license file if you like.

**Comment by d11e9** [comment]
Agreed. will remove asap

**Comment by d11e9** [comment]
@chriseth What full name, if any do you want copyright attributed to? (included in MIT License file)

**Pull request merged**